src/json/pull_parser.cr

```diff
@@ -87,20 +145,28 @@ class JSON::PullParser
     read_end_array
   end
 
+  # Reads the beginning of an object.
   def read_begin_object
     expect_kind :begin_object
     read_next
   end
 
+  # Reads the end of an object.
   def read_end_object
     expect_kind :end_object
     read_next
   end
 
+  # Reads an object's key and returns it.
   def read_object_key
     read_string
   end
 
+  # Reads a whole object.
+  #
+  # It reads the beginning of the object, yield each key and key location, and reads the end of the object.
```

Comment:
Maybe add that the values need to be consumed in the block, otherwise the pull parser breaks?